### PR TITLE
Avoid printing a stack trace when there is no resultsPath

### PR DESCRIPTION
**Pull request by aeisenberg**
I don't know exactly when this can happen, but a customer has just
shown me a stack trace like this:

```
TypeError: Cannot destructure property 'resultsPath' of 'resultsPaths' as it is undefined.
    at Object.interpretResults (/xxx/.vscode/extensions/github.vscode-codeql-1.4.5/out/query-results.js:120:13)
    at InterfaceManager._getInterpretedResults (/xxx/.vscode/extensions/github.vscode-codeql-1.4.5/out/interface.js:377:45)
    at InterfaceManager.showResultsAsDiagnostics (/xxx/.vscode/extensions/github.vscode-codeql-1.4.5/out/interface.js:447:43)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async InterfaceManager.handleMsgFromView (/xxx/.vscode/extensions/github.vscode-codeql-1.4.5/out/interface.js:151:29)
```

This commit will avoid printing this stack trace and instead print
a more descriptive message to the logs.

## Checklist

- [n/a] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [n/a] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-codeql` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/src/interface.ts [hunk 1]
```diff
@@ -561,24 +561,26 @@ export class InterfaceManager extends DisposableObject {
...
-  ): Promise<Interpretation> {
+  ): Promise<Interpretation | undefined> {
     const sarif = await interpretResults(
       this.cliServer,
       metadata,
       resultsPaths,
       sourceInfo
     );
+
+    if (!resultsPaths) {
```
Comment (adityasharad) [inline extensions/ql-vscode/src/interface.ts:572]: Should we do this before the call to `interpretResults` above? Or does that populate `resultsPaths`?

Comment (aeisenberg) [reply extensions/ql-vscode/src/interface.ts:572]: Yes, I think this should be moved up.

**Review by adityasharad — APPROVED** [review]
Looks reasonable. I assume the caller is prepared to handle an `undefined` response?

**Comment by aeisenberg** [comment]
> Looks reasonable. I assume the caller is prepared to handle an `undefined` response?

Yes.

**Pull request merged**